api/app/utils/fba_calculator.py

```diff
@@ -191,108 +186,133 @@ def get_approx_flame_length(head_fire_intensity: float):
     """
     return math.sqrt(head_fire_intensity / 300)
 
-# def get_critical_hours_start(critical_ffmc: float, solar_noon_ffmc: float, temperature: float,
-#                              relative_humidity: float, wind_speed: float, precip: float):
-#     """ Returns the hour of day (on 24H clock) at which the hourly FFMC crosses the
-#     threshold of critical_ffmc.
-#     Returns None if the hourly FFMC never reaches critical_ffmc.
-#     """
-#     if solar_noon_ffmc >= critical_ffmc:
-#         logger.info('Solar noon FFMC >= critical FFMC')
-#         # go back in time in increments of 0.5 hours
-#         clock_time = 13-0.5  # start from solar noon - 0.5 hours
-#         while get_hourly_ffmc_on_diurnal_curve(solar_noon_ffmc, clock_time, temperature,
-#                                                relative_humidity, wind_speed, precip) >= critical_ffmc:
-#             clock_time -= 0.5
-#             if clock_time == -0.5:
-#                 break
-#         # add back the half hour that caused FFMC to drop below critical_ffmc (or that
-#           pushed time below 0.0)
-#         clock_time += 0.5
-#         logger.info('%s', clock_time)
-#         return clock_time
-
-#     logger.info('Solar noon FFMC %s < critical FFMC %s', solar_noon_ffmc, critical_ffmc)
-#     # go forward in time in increments of 0.5 hours
-#     clock_time = 13 + 0.5  # start from solar noon + 0.5 hours
-#     while get_hourly_ffmc_on_diurnal_curve(solar_noon_ffmc, clock_time, temperature,
-#                                            relative_humidity, wind_speed, precip) < critical_ffmc:
-#         logger.info('Clock time %s has HFFMC %s', clock_time, get_hourly_ffmc_on_diurnal_curve(
-#             solar_noon_ffmc, clock_time, temperature, relative_humidity, wind_speed, precip))
-#         clock_time += 0.5
-#         if clock_time == 24.0:
-#             return None
-#     return clock_time
-
-
-# def get_critical_hours_end(critical_ffmc: float, solar_noon_ffmc: float, critical_hour_start: float,
-#                            temperature: float, relative_humidity: float, wind_speed: float, precip: float):
-#     """ Returns the hour of day (on 24H clock) at which the hourly FFMC drops below
-#     the threshold of critical_ffmc.
-#     Should only be called if critical_hour_start is not None.
-#     If diurnally-adjusted FFMC never drops below critical_ffmc in the day, will return 23.5 (11:30 pm).
+
+def get_afternoon_overnight_diurnal_ffmc(hour_of_interest: int, solar_noon_ffmc: float):
+    """ Returns the diurnal FFMC (an approximation) estimated for the given hour_of_interest,
+    based on the solar_noon_ffmc.
+    Hour_of_interest should be expressed in PDT time zone, and can only be between the hours
+    1330 and 0700 the next morning. Otherwise, must use different function.
+    """
+    with open('app/data/diurnalFFMC_redbook.xlsx', 'rb') as excel_file:
```

Comment:
Two concerns
1) xls is pretty heavy, I'd prefer we use csv or json to load this in - but whatever it works.
2) more critical - get_afternoon_overnight_diurnal_ffmc is being called multiple times, so the excel spreadsheet is being loaded over and over again - I'd want to see the data being loaded once.

Reply:
Do you think it would be better to split the excel file (which has 2 sheets in it) into 2 separate CSV files?

(also, right now we're only using 1 of the sheets)

Reply:
Yes, for a number of reasons
1) xls is heavy and slow (memory, load time and processor)
2) introducing another dependency when csv has built in support.

I'm actually more concerned about having yet another dependancy than anything else. BUT. It works. So why change it? Put a TODO in there and leave it at that.

Reply:
It may be worth making a singleton class that loads this file once with our existing Singleton implementation.

Reply:
Singleton class made! Nice and easy